warehouse/templates/search/results.html

```diff
@@ -143,7 +139,7 @@ <h2 class="no-top-padding">Filter Projects</h2>
           <div>
             <input id="search" type="hidden" name="q" value="{{ term }}">
             <label for="order">Order by &nbsp;</label>
-            <select class="-js-form-submit-trigger" name="o">
+            <select class="-js-form-submit-trigger" id="order" name="o">
```

Comment:
It correctly makes the label correspond to the form element via the `for` attribute.